src/modules/SuSEFirewallCMDLine.rb

```diff
@@ -1599,6 +1634,40 @@ def Run
       nil
     end
 
+  private
+    # Returns true if FirewallD is the running backend
+    def firewalld?
+      return true if SuSEFirewall.is_a?(Yast::SuSEFirewalldClass)
```

Comment:
this depends on hidden dependecy of SuSEFirewall on Yast::SuSEFirewalldClass, I prefer explicit requirement of Yast::SuSEFirewalldClass class for this comparison